lib/facility.rb

```diff
@@ -1,14 +1,189 @@
+require 'pry'
+require 'date'
+require 'faraday'
+require 'json'
+require './lib/dmv'
+require './lib/vehicle'
+require './lib/facility'
+require './lib/registrant'
+require './lib/vehicle_factory'
+require './lib/dmv_data_service'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
+
+  attr_accessor :name, :address, :phone, :services
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(hash)
+    @name = hash[:name]
+    @address = hash[:address]
+    @phone = hash[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+  #refactor
+  def register_vehicle(car)
+    if services.find {|service| service.include?('Vehicle Registration')}
+    current_time = DateTime.now
+    cdt = current_time.strftime "%d/%m/%Y %H:%M"
+    car.registration_date = cdt
+    @registered_vehicles << car
+
+      if Date.today.year - car.year > 25
+        car.plate_type = :antique
+      elsif Date.today.year - car.year <= 25 && car.engine == :ice
+        car.plate_type = :regular
+      else
+        car.plate_type = :ev
+      end
+
+      if car.plate_type == :antique
+        @collected_fees += 25
+      elsif car.engine == :ev
+        @collected_fees += 200
+      else
+        @collected_fees =+ 100
+      end
```

Comment:
you could probably assign collected feeds while going through your first conditional. 

```ruby
      if Date.today.year - car.year > 25
        car.plate_type = :antique
        @collected_fees += 25
      elsif Date.today.year - car.year <= 25 && car.engine == :ice
        car.plate_type = :regular
        @collected_fees += 100
      else
        car.plate_type = :ev
        @collected_fees += 200
      end
```